tests/test_flow/test_logging.py

```diff
@@ -2,17 +2,24 @@
 
 import logging
 import threading
+import bionic as bn
 
 
 class LogChecker:
     def __init__(self, caplog):
         self._caplog = caplog
 
-    def expect(self, *expected_messages):
+    def expect_all(self, *expected_messages):
         actual_messages = [record.getMessage() for record in self._caplog.records]
         assert set(actual_messages) == set(expected_messages)
         self._caplog.clear()
 
+    def contains_substrings(self, *substrings):
```

Comment:
nit: can we name this in a way that's congruent with the other method? Such as `expect_superset`?

Also, do we expect the number of substrings to match the number of logged messages? Or could there be extra messages that don't match anything?

Reply:
Renamed to `expect_regex`

> Also, do we expect the number of substrings to match the number of logged messages? Or could there be extra messages that don't match anything?

No, this is due to differences between running AIP in fake and real gcp. The message `Downloading ... from GCS` is logged in real GCP but not in fake. (Tried temporarily cleaning out `tmp_path` and doing a deep copy of the AIP job but it did not work, fixing this in fake GCP may be complicated.)